platforms/core-configuration/kotlin-dsl-tooling-builders/src/main/kotlin/org/gradle/kotlin/dsl/tooling/builders/internal/KotlinScriptingModelBuildersRegistrationAction.kt

```diff
@@ -37,13 +37,18 @@ class KotlinScriptingModelBuildersRegistrationAction : ProjectConfigureAction {
         builders.register(IsolatedScriptsModelBuilder)
 
         if (project.parent == null) {
-            val isolatedProjects = project.serviceOf<BuildModelParameters>().isIsolatedProjects
-            if (isolatedProjects) {
-                builders.register(IsolatedProjectsSafeKotlinDslScriptsModelBuilder(intermediateModelProvider))
+            val modelParameters = project.serviceOf<BuildModelParameters>()
+            val isolatedProjects = modelParameters.isIsolatedProjects
+            val builder = if (isolatedProjects) {
+                IsolatedProjectsSafeKotlinDslScriptsModelBuilder(intermediateModelProvider)
             } else {
-                builders.register(KotlinDslScriptsModelBuilder)
+                KotlinDslScriptsModelBuilder
             }
 
+            builders.register(builder)
+            if (modelParameters.isResilientModelBuilding) {
+                builders.register(ResilientKotlinDslScriptsModelBuilder(builder))
```

Comment:
💭 Once we will have new resilient "tryGetModel()" API, we can just use one model and register one builder